src/basic/macro.h

```diff
@@ -506,4 +506,23 @@ static inline int __coverity_check__(int condition) {
         DEFINE_PUBLIC_TRIVIAL_REF_FUNC(type, name);                    \
         DEFINE_PUBLIC_TRIVIAL_UNREF_FUNC(type, name, free_func);
 
+/* Negative return values from impl are mapped to EXIT_FAILURE, and
+ * everything else means success! */
+#define DEFINE_MAIN_FUNCTION(impl)                                      \
+        int main(int argc, char *argv[]) {                              \
+                int r;                                                  \
+                r = impl(argc, argv);                                   \
+                return r < 0 ? EXIT_FAILURE : EXIT_SUCCESS;             \
```

Comment:
Is it necessary to assign value for `r`?

Reply:
It's not. But I think this "verbose" version is more readable, and is also more similar to the other function. Dunno, I can change it if you think it's better, but since it's just in one place, I think I like this form better.

Reply:
OK. I think the current form is good enough.